.github/workflows/top-level.yml

```diff
@@ -0,0 +1,81 @@
+on:
```

Comment:
There is two new suggested yml file at 
https://analogdevicesinc.github.io/doctools/deploy.html#github-pages-deployment
that uses two new shared actions, `gh-pages-deploy@action` and `gh-pages-rm-path@action`.
The first deploys for commits in the main branch as usual, but also deploys for pull requests pull/<id> paths.
The second .yml is to then clean-up pull/<id> when it gets closed.